app/src/main/java/org/oppia/android/app/databinding/MarginBindingAdapters.java

```diff
@@ -29,36 +24,18 @@ public static void setLayoutMarginStart(@NonNull View view, float marginStart) {
   public static void setLayoutMarginEnd(@NonNull View view, float marginEnd) {
     if (view.getLayoutParams() instanceof MarginLayoutParams) {
       MarginLayoutParams params = (MarginLayoutParams) view.getLayoutParams();
-      float marginStart = params.getMarginStart();
-      if (isRtlLayout(view)) {
-        setLayoutDirectionalMargins(view, (int) marginEnd, (int) marginStart);
-      } else {
-        setLayoutDirectionalMargins(view, (int) marginStart, (int) marginEnd);
-      }
+      MarginLayoutParamsCompat.setMarginEnd(params, (int) marginEnd);
       view.requestLayout();
     }
   }
 
-  private static void setLayoutDirectionalMargins(
-      @NonNull View view,
-      int marginStart,
-      int marginEnd
-  ) {
-    MarginLayoutParams params = (MarginLayoutParams) view.getLayoutParams();
-    params.setMargins(marginStart, params.topMargin, marginEnd, params.bottomMargin);
-  }
-
   /** Used to set a margin-top for views. */
   @BindingAdapter("app:layoutMarginTop")
   public static void setLayoutMarginTop(@NonNull View view, float marginTop) {
     if (view.getLayoutParams() instanceof MarginLayoutParams) {
       MarginLayoutParams params = (MarginLayoutParams) view.getLayoutParams();
-      params.setMargins(
-          params.getMarginStart(),
-          (int) marginTop,
-          params.getMarginEnd(),
-          params.bottomMargin
-      );
+      params.topMargin = (int) marginTop;
+      view.setLayoutParams(params);
```

Comment:
I notice that we are setting the layout parameters of the view only with MarginTop and Bottom below. Why is this not the case for marginStart/End, and `layoutMargin` as well?

Reply:
I updated the PR description with an explanation

Reply:
The documentation doesn't say, so I can only guess: marginTop/Bottom are java fields, while marginStart/End are java method calls. In java, when a field is changed externally, the object doesn't know, thus it doesn't know it needs to recompute the view layout. On the other hand, method calls are more flexible. If you look at Android's source code, you can see that calling setMarginStart does more than just setting a field: https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/ViewGroup.java;l=8619?q=Marginlayoutparams%20setmarginstart&sq=&ss=android

Reply:
You're correct, according to the source code, setMarginStart/End returns the computed layout margins, therefore no need to call ViewGroup#setLayoutParams(LayoutParams) afterwards.

I think part of my confusion had to do with the fact that for start/end, we are relying oon MarginLayoutParamsCompat, but for top/bottom, we are using MarginLayoutParams. I have noticed that MarginLayoutParamsCompat does not provide means for setting the values of the top/bottom margins.

Reply:
As for my other comment:

> It seems that the main difference between this code and the one in the https://github.com/oppia/oppia-android/pull/4874 is the fact that we are calling view.setLayoutParams(params) before requesting the layout. It may be worth explaining in the PR description how or why this resolves the UI distortion issue.

I think it is now clear that the distortion in was caused by the fact that in #4874, we were not calling `ViewGroup#setLayoutParams()` after updating the margin values for top and bottom. You could just update the description to reflect this.